src/s2/util/math/exactfloat/bignum.h

```diff
@@ -0,0 +1,496 @@
+// Copyright 2025 Google LLC
+// Author: smcallis@google.com (Sean McAllister)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cmath>
+#include <cstdint>
+#include <limits>
+#include <optional>
+#include <type_traits>
+
+#include "absl/algorithm/container.h"
+#include "absl/base/attributes.h"
+#include "absl/container/inlined_vector.h"
+#include "absl/log/absl_check.h"
+#include "absl/numeric/bits.h"
+#include "absl/numeric/int128.h"
+#include "absl/strings/ascii.h"
+#include "absl/strings/str_format.h"
+
+namespace exactfloat_internal {
+
+// A class to support arithmetic on large, arbitrary precision integers.
+//
+// Large integers are represented as an array of uint64_t values.
+class Bignum {
+ public:
+  // Wrap uint64_t in a struct so we can make value-initialization a noop.
+  //
+  // Avoiding value-initialization overhead saves us 50% on some benchmarks.
```

Comment:
Can you be more specific? I want to be able to easily track this down.  Which benchmarks and why? InlinedVector?